pynxtools_xps/vms/vamas.py

```diff
@@ -38,26 +44,78 @@
     construct_entry_name,
     construct_data_key,
     construct_detector_data_key,
-    to_snake_case,
+    convert_pascal_to_snake,
     get_minimal_step,
+    check_for_allowed_in_list,
+    re_map_keys,
+    re_map_values,
+    drop_unused_keys,
+)
+from pynxtools_xps.value_mappers import (
+    convert_measurement_method,
+    convert_energy_type,
+    convert_energy_scan_mode,
 )
 
+EXP_MODES = [
+    "MAP",
+    "MAPDP",
+    "MAPSV",
+    "MAPSVDP",
+    "NORM",
+    "SDP",
+    "SDPSC",
+    "SEM",
+]
+
+ALLOWED_TECHNIQUES = [
+    "AES",
+    "AES diff",
+    "AES dir",
+    "EDX",
+    "ELS",
+    "FABMS",
+    "FABMS energy spec",
+    "ISS",
+    "SIMS",
+    "SIMS energy spec",
+    "SNMS",
+    "SNMS energy spec",
+    "UPS",
+    "XPS",
+    "XRF",
+]
+
 UNITS: dict = {
-    "instrument/sample_normal_polarangle_tilt": "degree ",
-    "instrument/sample_tilt_azimuth": "degree",
-    "instrument/sample_rotation_angle": "degree",
-    "source/source_analyzer_angle": "degree",
-    "source/excitation_energy": "eV",
-    "source/particle_charge": "C",
-    "analyser/analyzer_take_off_azimuth": "degree",
-    "analyser/analyzer_take_off_polar": "degree",
+    "beam_xray/excitation_energy": "eV",
+    "beam_xray/extent": "micro-m",
+    "beam_xray/particle_charge": "C",
+    "beam_xray/source_beam_width_x": "micro-m",
+    "beam_xray/source_beam_width_y": "micro-m",
+    "beam_xray/source_analyser_angle": "degree",
+    "source_xray/source_azimuth": "degree",
+    "source_xray/source_power": "W",
+    "source_sputter/sputter_ion_charge": "C",
+    "source_sputter/sputter_source_energy": "eV",
+    "source_sputter/sputter_source_beam_current": "A",
+    "source_sputter/sputter_source_width_x": "micro-m",
+    "source_sputter/sputter_source_width_y": "micro-m",
+    "source_sputter/sputter_source_incidence_polar_angle": "degree",
+    "source_sputter/sputter_source_azimuth_angle": "degree",
+    "analyser/analyser_take_off_azimuth": "degree",
+    "analyser/analyser_take_off_polar": "degree",
     "analyser/analysis_width_x": "m",
     "analyser/analysis_width_y": "m",
     "analyser/target_bias": "V",
     "analyser/time_correction": "s",
     "analyser/work_function": "eV",
+    "collectioncolumn/spatial_acceptance": "micro-m",
     "energydispersion/pass_energy": "eV",
+    "energydispersion/differential_width_aes": "eV",
     "detector/dwell_time": "s",
+    "sample/sample_tilt_normal_polar": "degree ",
+    "sample/sample_tilt_azimuth": "degree",
+    "sample/sample_rotation": "degree",
     "data/start_energy": "eV",
     "data/step_size": "eV",
```

Comment:
Be confirm that the units here follow the same nomenclature as nomad.

Reply:
Should be ok according to the defaults units in pint.